src/ws/session.c

```diff
@@ -515,12 +516,7 @@ open_session (pam_handle_t *pamh)
       return res;
     }
 
-  /* Yes, buf "leaks" */
-  buf = malloc (sizeof (struct passwd) + 8192);
-  if (buf == NULL)
-    res = ENOMEM;
-  else
-    res = getpwnam_r (name, buf, (char *)(buf + 1), 8192, &pwd);
+  res = getpwnam_r (name, &pwd_buf, &pwd_string_buf[0], sizeof(pwd_string_buf), &pwd);
```

Comment:
Much nicer, thanks for the cleanup!

/&pwd_string_buf[0]/pwd_string_buf/ and our code style wants  a space in the `sizeof(`.